conductr_cli/bndl_create.py

```diff
@@ -38,92 +48,167 @@ def bndl_create(args):
+                    bundle_conf = ConfigFactory.parse_string(bundle_conf_data.decode('UTF-8'))
 
-        if not has_oci_layout:
-            log.error('bndl: Invalid OCI Image. Missing oci-layout')
+                    if 'name' in bundle_conf:
+                        args.name = bundle_conf['name']
+                except:
+                    pass  # ignore exceptions - we'll catch the bad config below
 
-            return 2
+        mtime = first_mtime(input_dir, SHAZAR_TIMESTAMP_MIN) if mtime is None else mtime
 
-        oci_manifest, oci_config = bndl_oci.oci_image_extract_manifest_config(oci_image_dir, args.tag)
-        bundle_conf = oci_image_bundle_conf(args, component_name, oci_manifest, oci_config)
-        bundle_conf_name = os.path.join(args.name, 'bundle.conf')
-        bundle_conf_data = bundle_conf.encode('UTF-8')
+        if process_oci:
+            has_oci_layout = os.path.isfile(os.path.join(component_dir, 'oci-layout'))
 
-        # bundle data timestamps can vary based on how it's acquired (docker save, our own resolver, etc) so we
-        # deterministically set the mtime of the files to be based on the OCI config 'created' value if available
+            refs_dir = os.path.join(component_dir, 'refs')
 
-        if oci_config and 'created' in oci_config:
-            mtime = arrow.get(oci_config['created']).timestamp
-        else:
-            mtime = first_mtime(temp_dir)
+            if args.image_tag is None and os.path.isdir(refs_dir):
+                for ref in os.listdir(refs_dir):
+                    args.image_tag = ref
+                    break
+
+            ref_exists = args.image_tag is not None and os.path.isfile(os.path.join(refs_dir, args.image_tag))
+
+            if not ref_exists:
+                log.error('bndl: Invalid OCI Image. Cannot find requested tag "{}" in OCI Image'.format(args.image_tag))
+
+                return 2
+
+            if not has_oci_layout:
+                log.error('bndl: Invalid OCI Image. Missing oci-layout')
+
+                return 2
+
+            oci_manifest, oci_config = bndl_oci.oci_image_extract_manifest_config(component_dir, args.image_tag)
+            bundle_conf = oci_image_bundle_conf(args, component_name, oci_manifest, oci_config)
+            bundle_conf_data = bundle_conf.encode('UTF-8')
+
+            # bundle data timestamps can vary based on how it's acquired (docker save, our own resolver, etc) so we
+            # deterministically set the mtime of the files to be based on the OCI config 'created' value if available
+
+            if oci_config and 'created' in oci_config:
+                mtime = arrow.get(oci_config['created']).timestamp
+
+        try:
+            bundle_conf = ConfigFactory.parse_string(bundle_conf_data.decode('UTF-8'))
+        except ConfigException:
+            log.error('bndl: Unable to parse bundle.conf')
+            return 1
+
+        load_bundle_args_into_conf(bundle_conf, args, with_defaults=False)
```

Comment:
This is where we do the `bundle.conf` parsing to update any values from the provided arguments.